services/relationservice.js

```diff
@@ -309,17 +311,29 @@ module.exports = class RelationService {
             //confirm to delete
             if (res) {
               this.getEditingService().getCurrentWorkflowData().session.pushDelete(this._relationLayerId, relationfeature);
-              this.relations.splice(index, 1);     // remove feature from relation features
-              this.tools.splice(index, 1); // remove tool from relation tools
+              this.relations.splice(index, 1); // remove feature from relation features
+              this.tools.splice(index, 1);     // remove tool from relation tools
               this.getEditingService().removeRelationLayerUniqueFieldValuesFromFeature({
                 layerId: this._relationLayerId,
                 relationLayerId: this.parent.layerId,
                 feature: relationfeature
               });
-              //@TODO Need to handle deletion of new relation feature (not yet committed) check saveAll button status
-              // relationfeature.isNew())
               featurestore.removeFeature(relationfeature);
-              this.updateParentWorkflows();
+              // check if relation feature delete is new.
+              // In this case, we need to check if there are temporary changes not related to this current feature
+              if (
+                  relationfeature.isNew()
+                  && !WorkflowsStack
+                    ._workflows
+                    .reduce((a,w) => a || w.getSession()._temporarychanges.filter(({ feature }) => relationfeature.getUid() !== feature.getUid()).length > 0, false)
```

Comment:
@volterra79 in such cases `Array.Some` it could be more straightforward:

```suggestion
                  && !WorkflowsStack
                     ._workflows
                     .some(w => w.getSession()._temporarychanges.some(({ feature }) => relationfeature.getUid() !== feature.getUid()))
```

Reply:
@Raruto fix here https://github.com/g3w-suite/g3w-client-plugin-editing/pull/110/commits/fe59b7b899b40c521d81f53f33405d80592d7a11

Reply:
@volterra79 There is something weird:

0. start editing from queryresults

![image](https://github.com/g3w-suite/g3w-client-plugin-editing/assets/9614886/2066d8a9-42ed-43a8-aa07-16f8fcb0db57)

1. Create and link a new feature relation (plus icon)
2. click on the save all icon (floppy disk)
3. try delete the new inserted feature (nothing happens)

![image](https://github.com/g3w-suite/g3w-client-plugin-editing/assets/9614886/5ec40910-0fb5-4804-98b6-a5f9796c237d)